src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,759 @@
+  arrayView1d< real64 > const p_n = nodeManager.getExtrinsicData< extrinsicMeshData::Pressure_n >();
+  arrayView1d< real64 > const p_np1 = nodeManager.getExtrinsicData< extrinsicMeshData::Pressure_np1 >();
+
+  ArrayOfArraysView< localIndex const > const faceToNodeMap = faceManager.nodeList().toViewConst();
+
+  /// array of indicators: 1 if a face is on on free surface; 0 otherwise
+  arrayView1d< localIndex > const freeSurfaceFaceIndicator = faceManager.getExtrinsicData< extrinsicMeshData::FreeSurfaceFaceIndicator >();
+
+  /// array of indicators: 1 if a node is on on free surface; 0 otherwise
+  arrayView1d< localIndex > const freeSurfaceNodeIndicator = nodeManager.getExtrinsicData< extrinsicMeshData::FreeSurfaceNodeIndicator >();
+
+  fsManager.apply( time,
+                   domain,
+                   "faceManager",
+                   string( "FreeSurface" ),
+                   [&]( FieldSpecificationBase const & bc,
+                        string const &,
+                        SortedArrayView< localIndex const > const & targetSet,
+                        Group &,
+                        string const & )
+  {
+    string const & functionName = bc.getFunctionName();
+
+    if( functionName.empty() || functionManager.getGroup< FunctionBase >( functionName ).isFunctionOfTime() == 2 )
+    {
+      real64 const value = bc.getScale();
+
+      freeSurfaceFaceIndicator.setValues< serialPolicy >( 0 );
+      freeSurfaceNodeIndicator.setValues< serialPolicy >( 0 );
+
+      for( localIndex i = 0; i < targetSet.size(); ++i )
+      {
+        localIndex const kf = targetSet[ i ];
+        freeSurfaceFaceIndicator[kf] = 1;
+
+        localIndex const numNodes = faceToNodeMap.sizeOfArray( kf );
+        for( localIndex a=0; a < numNodes; ++a )
+        {
+          localIndex const dof = faceToNodeMap( kf, a );
+          freeSurfaceNodeIndicator[dof] = 1;
+
+          p_np1[dof] = value;
+          p_n[dof]   = value;
+          p_nm1[dof] = value;
+        }
+      }
+    }
+    else
+    {
+      GEOSX_ERROR( "This option is not supported yet" );
+    }
+  } );
+}
+
+
+
+real64 AcousticWaveEquationSEM::solverStep( real64 const & time_n,
+                                            real64 const & dt,
+                                            integer const cycleNumber,
+                                            DomainPartition & domain )
+{
+  return explicitStep( time_n, dt, cycleNumber, domain );
+}
+
+real64 AcousticWaveEquationSEM::evaluateRicker( real64 const & time_n, real64 const & f0 )
+{
+  real64 o_tpeak = 1.0/f0;
+  real64 pulse = 0.0;
+  if((time_n <= -0.9*o_tpeak) || (time_n >= 2.9*o_tpeak))
+    return pulse;
+
+  real64 const pi = 2.0*acos( 0 );
+  real64 const tmp = f0*time_n-1.0;
+  pulse = -(time_n-1)*exp( -2*(tmp*pi)*(tmp*pi) );
+
+  return pulse;
+}
+
+
+real64 AcousticWaveEquationSEM::evaluateRickerOrder1( real64 const & time_n, real64 const & f0 )
```

Comment:
Also all of these Ricker functions share most of their code. I'd either unify them by adding an `int` parameter that specified the order. I don't think this will be a bottleneck so it shouldn't matter, and it will greatly simplify the dispatch.

Reply:
Yes, I merged the three function with a parameter that specified the order now.